Sources/Protocols/Nibable.swift

```diff
@@ -28,13 +28,9 @@ public extension Nibable where Self: UIView {
 
     init(owner: AnyObject? = nil) {
         let views = Self.nib.instantiate(withOwner: owner, options: [:])
-        for view in views {
-            if let view = view as? Self {
-                self = view
-                return
-            }
-        }
-        fatalError("Nib for class \(Self.nibName) cound not be loaded!")
+        self = views.lazy.compactMap { $0 as? Self }.first ?? { () -> Self in
```

Comment:
Hello @mkj-is 
In this part of code you want to find the first view of class `Self` and set it for `self`. Yes, probably it will be better to use something like `Array.first(where:)` here, but in such case we will need to use a typecasting for the result value. So, the only one better solution that I can see now is to create a special version of first(where:), something like that:
```
extension Array {
    func firstOfType<ReturnType>(where predicate: (Element) throws -> Bool) rethrows -> ReturnType? {
        return try self.first(where: predicate) as? ReturnType
    }
}
```

Reply:
Agreed, you described it well and I see the reasons now and I do not want to add one time extension.